core/src/main/java/org/mule/runtime/core/api/exception/SystemExceptionHandler.java

```diff
@@ -25,6 +27,14 @@
    * @param exception which occurred
    */
   void handleException(Exception exception);
+
+  /**
+   * Take some action when a system exception has occurred (i.e., there was no message in play when the exception occurred).
+   *
+   * @param exception which occurred
+   * @param componentLocation the {@link ComponentLocation} which produced the exception
+   */
+  void handleException(Exception exception, ComponentLocation componentLocation);
```

Comment:
We cannot add a new method to an existing  API interface, not at least without a default implementation that makes sense.